From ffc701b84a39877abc98e8112bf28b5dcaa8e399 Mon Sep 17 00:00:00 2001
From: Eugene Susla <eugenesusla@google.com>
Date: Fri, 4 Dec 2020 16:01:46 -0800
Subject: [PATCH] Allow CDM to hide overlays

Since CDM has sensitive user consent UIs, it should be able to hide
non-system overlays

Test: use a 3p overlay app with a visible overlay to ensure overlay disappears when CDM is shown
Bug: 171221090
Change-Id: I78673b7b95f4c3cb31fab180201eb17c9123bddf
Merged-In: I78673b7b95f4c3cb31fab180201eb17c9123bddf
(cherry picked from commit f717d7195fa8654a7c9e4e28bbe473f8513eef64)
---
 packages/CompanionDeviceManager/AndroidManifest.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/CompanionDeviceManager/AndroidManifest.xml b/packages/CompanionDeviceManager/AndroidManifest.xml
index ea9b52cbf3d5..e4e5b9fa781a 100644
--- a/packages/CompanionDeviceManager/AndroidManifest.xml
+++ b/packages/CompanionDeviceManager/AndroidManifest.xml
@@ -31,6 +31,7 @@
     <uses-permission android:name="android.permission.CHANGE_WIFI_STATE"/>
     <uses-permission android:name="android.permission.RADIO_SCAN_WITHOUT_LOCATION"/>
     <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES"/>
+    <uses-permission android:name="android.permission.HIDE_NON_SYSTEM_OVERLAY_WINDOWS"/>

     <application
         android:allowClearUserData="true"
--
2.29.0
